### PR TITLE
Fix #180:  Highfi Hint solution Portrait

### LINKED ISSUE #180: Add support for hints in the exploration player
For parity with Oppia web, we need to add support for hints in the exploration player.

### PR DESCRIPTION
**Pull request by veena14cs**
## Explanation
This PR aims at fixing hifi of hints and solution in Portrait mode.

### Mocks
https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/a70b76a7-49ae-47b6-b5fe-f148b71ee620/LM-Q5-HS-Hint-Dialog-Reveal-Hint-4-Hint-1-Opened-
https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/d43c5d96-945a-4b89-9552-c4fb482ce2f2/LM-Q5-HS-Hint-Dialog-Reveal-Hint-4-

### Screenshot
Screenshot_20200414-123146
Screenshot_20200414-123152

Screenshot_20200414-170316

### Accessibility Scanner

Screenshot_20200414-122315
Screenshot_20200414-122444

### CODE REVIEW

File: app/src/main/res/layout/hints_and_solution_fragment.xml [hunk 1]
```diff
@@ -33,24 +44,23 @@
...
       <androidx.recyclerview.widget.RecyclerView
         android:id="@+id/hints_and_solution_recycler_view"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:clipToPadding="false"
         android:overScrollMode="never"
-        android:paddingTop="8dp"
-        android:paddingBottom="128dp"
+        android:paddingTop="20dp"
+        android:paddingBottom="168dp"
```
Comment (rt4914) [inline app/src/main/res/layout/hints_and_solution_fragment.xml:56]: This is inconsistent in the mocks meaning, it is `152dp` in mock but I remember in various other mocks it is `168dp`. So you can finalise this what should it be finalise my suggestion is `168dp` to keep it consistent.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Suggested some changes.

File: app/src/main/res/layout/hints_and_solution_fragment.xml (hunk 1 above)
Comment (veena14cs) [reply app/src/main/res/layout/hints_and_solution_fragment.xml:56]: Okay, keeping it 168dp.

File: app/src/main/res/layout/hints_summary.xml [hunk 2]
```diff
@@ -10,108 +10,109 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:minWidth="160dp"
+        android:minHeight="48dp"
+        android:layout_marginStart="8dp"
+        android:layout_marginTop="16dp"
+        android:layout_marginBottom="32dp"
+        android:layout_marginEnd="8dp"
+        android:background="@drawable/reveal_hint_button_blue_background"
+        android:padding="4dp"
```
Comment (nikitamarysolomanpvt) [inline app/src/main/res/layout/hints_summary.xml]: Padding is not matching mock.

File: app/src/main/res/layout/hints_and_solution_fragment.xml [hunk 3]
```diff
@@ -33,24 +44,23 @@
...
 
       <androidx.recyclerview.widget.RecyclerView
         android:id="@+id/hints_and_solution_recycler_view"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:clipToPadding="false"
         android:overScrollMode="never"
-        android:paddingTop="8dp"
-        android:paddingBottom="128dp"
+        android:paddingTop="20dp"
```
Comment (nikitamarysolomanpvt) [inline app/src/main/res/layout/hints_and_solution_fragment.xml:55]: Top padding when added with item content padding is not matching

File: app/src/main/res/layout/solution_summary.xml [hunk 4]
```diff
@@ -11,119 +11,156 @@
...
         <TextView
-          android:id="@+id/solution_correct_answer"
+          android:id="@+id/solution_summary_label"
           android:layout_width="wrap_content"
           android:layout_height="wrap_content"
-          android:fontFamily="sans-serif-medium"
+          android:layout_gravity="start"
+          android:fontFamily="sans-serif"
+          android:layout_marginTop="12dp"
+          android:textStyle="bold"
```
Comment (nikitamarysolomanpvt) [inline app/src/main/res/layout/solution_summary.xml]: May i know why this is required?

**Review by nikitamarysolomanpvt — CHANGES_REQUESTED** [review]
PTAL on suggestions.

**Review by rt4914 — APPROVED** [review]
LGTM, just clear Nikita's comment and its good to go.

File: app/src/main/res/layout/hints_summary.xml (hunk 2 above)
Comment (veena14cs) [reply app/src/main/res/layout/hints_summary.xml]: Removed this padding.

File: app/src/main/res/layout/hints_and_solution_fragment.xml (hunk 3 above)
Comment (veena14cs) [reply app/src/main/res/layout/hints_and_solution_fragment.xml:55]: I removed margin top from the item and added bottom margin to divider to match.

File: app/src/main/res/layout/solution_summary.xml (hunk 4 above)
Comment (veena14cs) [reply app/src/main/res/layout/solution_summary.xml]: I tried to make it san-serif bold by using android font-family option. I removed it and used san-serif-black.

**Review by nikitamarysolomanpvt — APPROVED** [review]
LGTM

**Pull request merged**